src/config/configStore.ts

```diff
@@ -128,6 +129,11 @@ export abstract class BaseConfigStore<T extends BaseConfigStore.Options> extends
    * @param key The key.
    */
   public unset(key: string): boolean {
+    const property = Config.getAllowedProperties().find(allowedProp => allowedProp.key === key);
```

Comment:
It feels weird to have this here but not in any of the other methods. What was the big rationale for moving this here? To keep it out of the config plugin?

Reply:
Yes. Also config.set does this from within core (in config.ts) so I thought it would make sense to have a similar check for config.unset as well